AssertionError from apprise.notify(title=str(title), body=str(body))

### Before submitting a bug report...

- [X] This bug wasn't already reported.
(I have checked every bug report on GitHub)

- [X] I've cleared the sessions folder.


### Branch

master

### Commit

54e00cb

### Describe the bug

An AssertionError is being raised from file "d:\MS-Rewards-Farmer\src\utils.py", line 62, in sendNotification.

### Copy and paste your error

2024-08-25 13:27:19,062 [INFO] ********************[I hid my email here]********************
2024-08-25 13:27:19,062 [DEBUG] in __init__

DevTools listening on ws://127.0.0.1:50852/devtools/browser/20126ca2-5c80-45b7-a757-8d493f0db626
2024-08-25 13:27:28,216 [INFO] Screen size: 2307x1319
2024-08-25 13:27:28,223 [INFO] Device size: 2252x1168
2024-08-25 13:27:28,257 [DEBUG] out __init__
2024-08-25 13:27:28,263 [DEBUG] in __enter__
2024-08-25 13:27:41,653 [INFO] [LOGIN] Logging-in...
2024-08-25 13:27:41,675 [INFO] [LOGIN] Entering email...
2024-08-25 13:27:52,072 [DEBUG] isPasswordless = False
2024-08-25 13:27:52,086 [INFO] [LOGIN] Entering password...
2024-08-25 13:28:03,322 [DEBUG] isDeviceAuthEnabled = False
2024-08-25 13:28:03,330 [DEBUG] isTOTPEnabled = True
2024-08-25 13:28:03,331 [INFO] [LOGIN] Entering OTP...
2024-08-25 13:28:12,736 [DEBUG] isAskingToProtect = False
2024-08-25 13:28:12,747 [INFO] [LOGIN] Logged-in successfully !
2024-08-25 13:28:16,015 [INFO] [POINTS] You have 0.00 points on your account
2024-08-25 13:28:16,015 [INFO] [DAILY SET] Trying to complete the Daily Set...
2024-08-25 13:28:18,266 [INFO] [DAILY SET] Exiting
2024-08-25 13:28:18,266 [INFO] [PUNCH CARDS] Trying to complete the Punch Cards...
2024-08-25 13:28:22,434 [INFO] [PUNCH CARDS] Exiting
2024-08-25 13:28:22,434 [INFO] [MORE PROMOS] Trying to complete More Promotions...
2024-08-25 13:28:24,372 [DEBUG] promotionTitle=You’ve earned enough points to redeem!
2024-08-25 13:28:24,372 [DEBUG] Already done, continuing
2024-08-25 13:28:24,372 [DEBUG] promotionTitle=Do you know the answer?
2024-08-25 13:28:24,373 [DEBUG] Already done, continuing
2024-08-25 13:28:24,373 [DEBUG] promotionTitle=Expand your vocabulary
2024-08-25 13:28:48,150 [DEBUG] promotionTitle=Stay on top of the elections
2024-08-25 13:28:48,150 [DEBUG] Already done, continuing
2024-08-25 13:28:48,150 [DEBUG] promotionTitle=What time is it?
2024-08-25 13:29:13,761 [DEBUG] promotionTitle=Earn More Rewards points on mobile
2024-08-25 13:29:13,761 [DEBUG] Already done, continuing
2024-08-25 13:29:13,761 [DEBUG] promotionTitle=Collect them all
2024-08-25 13:29:13,761 [DEBUG] Already done, continuing
2024-08-25 13:29:15,234 [DEBUG] in __exit__ exc_type=<class 'AssertionError'> exc_value= traceback=<traceback object at 0x00000182B72D0680>
2024-08-25 13:29:15,479 [ERROR] 
Traceback (most recent call last):
  File "d:\MS-Rewards-Farmer\main.py", line 40, in main
    earned_points = executeBot(currentAccount, args)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\MS-Rewards-Farmer\main.py", line 245, in executeBot
    MorePromotions(desktopBrowser).completeMorePromotions()
  File "d:\MS-Rewards-Farmer\src\morePromotions.py", line 123, in completeMorePromotions
    Utils.sendNotification("Incomplete promotions(s)", incompletePromotions)
  File "d:\MS-Rewards-Farmer\src\utils.py", line 62, in sendNotification
    assert apprise.notify(title=str(title), body=str(body))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError
2024-08-25 13:29:15,482 [ERROR]
Traceback (most recent call last):
  File "d:\MS-Rewards-Farmer\main.py", line 40, in main
    earned_points = executeBot(currentAccount, args)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\MS-Rewards-Farmer\main.py", line 245, in executeBot
    MorePromotions(desktopBrowser).completeMorePromotions()
  File "d:\MS-Rewards-Farmer\src\morePromotions.py", line 123, in completeMorePromotions
    Utils.sendNotification("Incomplete promotions(s)", incompletePromotions)
  File "d:\MS-Rewards-Farmer\src\utils.py", line 62, in sendNotification
    assert apprise.notify(title=str(title), body=str(body))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "d:\MS-Rewards-Farmer\main.py", line 356, in <module>
    main()
  File "d:\MS-Rewards-Farmer\main.py", line 43, in main
    Utils.sendNotification(
  File "d:\MS-Rewards-Farmer\src\utils.py", line 62, in sendNotification
    assert apprise.notify(title=str(title), body=str(body))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError
Traceback (most recent call last):
  File "d:\MS-Rewards-Farmer\main.py", line 40, in main
    earned_points = executeBot(currentAccount, args)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\MS-Rewards-Farmer\main.py", line 245, in executeBot
    MorePromotions(desktopBrowser).completeMorePromotions()
  File "d:\MS-Rewards-Farmer\src\morePromotions.py", line 123, in completeMorePromotions
    Utils.sendNotification("Incomplete promotions(s)", incompletePromotions)
  File "d:\MS-Rewards-Farmer\src\utils.py", line 62, in sendNotification
    assert apprise.notify(title=str(title), body=str(body))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "d:\MS-Rewards-Farmer\main.py", line 356, in <module>
    main()
  File "d:\MS-Rewards-Farmer\main.py", line 43, in main
    Utils.sendNotification(
  File "d:\MS-Rewards-Farmer\src\utils.py", line 62, in sendNotification
    assert apprise.notify(title=str(title), body=str(body))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "d:\MS-Rewards-Farmer\main.py", line 359, in <module>
    Utils.sendNotification(
  File "d:\MS-Rewards-Farmer\src\utils.py", line 62, in sendNotification
    assert apprise.notify(title=str(title), body=str(body))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError

### Screenshots

![image](https://github.com/user-attachments/assets/917180f6-4106-4fb6-951e-23b32d08c7bb)

### Value of dashboard variable

https://gist.github.com/HaveSomeRubicon/f9a83517138126ab59b42835220ce175

Sorry, it worked for a bit longer than usual and I closed the issue out of excitement. I got the error again but this time it was followed by a bunch of index errors during punch cards and promotions. Here is the full log.

`2024-08-26 09:23:49,037 [INFO] ********************email hidden********************
2024-08-26 09:23:49,037 [DEBUG] in __init__

DevTools listening on ws://127.0.0.1:61406/devtools/browser/3de90db0-3b8c-4f66-8848-d62e113b0789
2024-08-26 09:23:58,979 [INFO] Screen size: 2307x1319
2024-08-26 09:23:58,979 [INFO] Device size: 2252x1168
2024-08-26 09:23:58,997 [DEBUG] out __init__
2024-08-26 09:23:59,000 [DEBUG] in __enter__
2024-08-26 09:24:02,738 [INFO] [LOGIN] Already logged-in
2024-08-26 09:24:10,204 [INFO] [POINTS] You have 2,252.00 points on your account
2024-08-26 09:24:10,204 [INFO] [DAILY SET] Trying to complete the Daily Set...
2024-08-26 09:24:25,111 [ERROR] [DAILY SET] Error Daily Set of card 1
Traceback (most recent call last):
  File "D:\MS-Rewards-Farmer\src\dailySet.py", line 27, in completeDailySet
    self.activities.openDailySetActivity(cardId)
  File "D:\MS-Rewards-Farmer\src\activities.py", line 22, in openDailySetActivity
    self.browser.utils.switchToNewTab(timeToWait=8)
  File "D:\MS-Rewards-Farmer\src\utils.py", line 216, in switchToNewTab
    self.webdriver.switch_to.window(window_name=self.webdriver.window_handles[1])
                                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^
IndexError: list index out of range
2024-08-26 09:24:31,086 [ERROR] [DAILY SET] Error Daily Set of card 2
Traceback (most recent call last):
  File "D:\MS-Rewards-Farmer\src\dailySet.py", line 27, in completeDailySet
    self.activities.openDailySetActivity(cardId)
  File "D:\MS-Rewards-Farmer\src\activities.py", line 22, in openDailySetActivity
    self.browser.utils.switchToNewTab(timeToWait=8)
  File "D:\MS-Rewards-Farmer\src\utils.py", line 216, in switchToNewTab
    self.webdriver.switch_to.window(window_name=self.webdriver.window_handles[1])
                                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^
IndexError: list index out of range
2024-08-26 09:24:37,320 [ERROR] [DAILY SET] Error Daily Set of card 3
Traceback (most recent call last):
  File "D:\MS-Rewards-Farmer\src\dailySet.py", line 27, in completeDailySet
    self.activities.openDailySetActivity(cardId)
  File "D:\MS-Rewards-Farmer\src\activities.py", line 22, in openDailySetActivity
    self.browser.utils.switchToNewTab(timeToWait=8)
  File "D:\MS-Rewards-Farmer\src\utils.py", line 216, in switchToNewTab
    self.webdriver.switch_to.window(window_name=self.webdriver.window_handles[1])
                                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^
IndexError: list index out of range
2024-08-26 09:24:43,145 [INFO] [DAILY SET] Exiting
2024-08-26 09:24:43,145 [INFO] [PUNCH CARDS] Trying to complete the Punch Cards...
2024-08-26 09:24:57,213 [INFO] [PUNCH CARDS] Exiting
2024-08-26 09:24:57,213 [INFO] [MORE PROMOS] Trying to complete More Promotions...
2024-08-26 09:25:03,201 [DEBUG] promotionTitle=You’ve earned enough points to redeem!
2024-08-26 09:25:03,201 [DEBUG] Already done, continuing
2024-08-26 09:25:03,201 [DEBUG] promotionTitle=The roots of hip hop
2024-08-26 09:25:32,668 [DEBUG] promotionTitle=Marvel at jellylfish
2024-08-26 09:25:33,066 [ERROR] [MORE PROMOS] Error More Promotions
Traceback (most recent call last):
  File "D:\MS-Rewards-Farmer\src\morePromotions.py", line 39, in completeMorePromotions
    self.activities.openMorePromotionsActivity(
  File "D:\MS-Rewards-Farmer\src\activities.py", line 29, in openMorePromotionsActivity
    self.browser.utils.switchToNewTab(timeToWait=5)
  File "D:\MS-Rewards-Farmer\src\utils.py", line 216, in switchToNewTab
    self.webdriver.switch_to.window(window_name=self.webdriver.window_handles[1])
                                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^
IndexError: list index out of range
2024-08-26 09:25:36,645 [DEBUG] promotionTitle=Complete this puzzle
2024-08-26 09:25:36,723 [ERROR] [MORE PROMOS] Error More Promotions
Traceback (most recent call last):
  File "D:\MS-Rewards-Farmer\src\morePromotions.py", line 39, in completeMorePromotions
    self.activities.openMorePromotionsActivity(
  File "D:\MS-Rewards-Farmer\src\activities.py", line 29, in openMorePromotionsActivity
    self.browser.utils.switchToNewTab(timeToWait=5)
  File "D:\MS-Rewards-Farmer\src\utils.py", line 216, in switchToNewTab
    self.webdriver.switch_to.window(window_name=self.webdriver.window_handles[1])
                                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^
IndexError: list index out of range
2024-08-26 09:25:40,096 [DEBUG] promotionTitle=Expand your vocabulary
2024-08-26 09:25:40,121 [ERROR] [MORE PROMOS] Error More Promotions
Traceback (most recent call last):
  File "D:\MS-Rewards-Farmer\src\morePromotions.py", line 39, in completeMorePromotions
    self.activities.openMorePromotionsActivity(
  File "D:\MS-Rewards-Farmer\src\activities.py", line 29, in openMorePromotionsActivity
    self.browser.utils.switchToNewTab(timeToWait=5)
  File "D:\MS-Rewards-Farmer\src\utils.py", line 216, in switchToNewTab
    self.webdriver.switch_to.window(window_name=self.webdriver.window_handles[1])
                                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^
IndexError: list index out of range
2024-08-26 09:25:43,438 [DEBUG] promotionTitle=Stay on top of the elections
2024-08-26 09:25:43,438 [DEBUG] Already done, continuing
2024-08-26 09:25:43,438 [DEBUG] promotionTitle=What time is it?
2024-08-26 09:25:43,535 [ERROR] [MORE PROMOS] Error More Promotions
Traceback (most recent call last):
  File "D:\MS-Rewards-Farmer\src\morePromotions.py", line 39, in completeMorePromotions
    self.activities.openMorePromotionsActivity(
  File "D:\MS-Rewards-Farmer\src\activities.py", line 29, in openMorePromotionsActivity
    self.browser.utils.switchToNewTab(timeToWait=5)
  File "D:\MS-Rewards-Farmer\src\utils.py", line 216, in switchToNewTab
    self.webdriver.switch_to.window(window_name=self.webdriver.window_handles[1])
                                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^
IndexError: list index out of range
2024-08-26 09:25:46,834 [DEBUG] promotionTitle=Vegan sushi delights
2024-08-26 09:25:46,877 [ERROR] [MORE PROMOS] Error More Promotions
Traceback (most recent call last):
  File "D:\MS-Rewards-Farmer\src\morePromotions.py", line 39, in completeMorePromotions
    self.activities.openMorePromotionsActivity(
  File "D:\MS-Rewards-Farmer\src\activities.py", line 29, in openMorePromotionsActivity
    self.browser.utils.switchToNewTab(timeToWait=5)
  File "D:\MS-Rewards-Farmer\src\utils.py", line 216, in switchToNewTab
    self.webdriver.switch_to.window(window_name=self.webdriver.window_handles[1])
                                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^
IndexError: list index out of range
2024-08-26 09:25:50,063 [DEBUG] promotionTitle=A step-by-step guide
2024-08-26 09:25:50,102 [ERROR] [MORE PROMOS] Error More Promotions
Traceback (most recent call last):
  File "D:\MS-Rewards-Farmer\src\morePromotions.py", line 39, in completeMorePromotions
    self.activities.openMorePromotionsActivity(
  File "D:\MS-Rewards-Farmer\src\activities.py", line 29, in openMorePromotionsActivity
    self.browser.utils.switchToNewTab(timeToWait=5)
  File "D:\MS-Rewards-Farmer\src\utils.py", line 216, in switchToNewTab
    self.webdriver.switch_to.window(window_name=self.webdriver.window_handles[1])
                                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^
IndexError: list index out of range
2024-08-26 09:25:53,047 [DEBUG] promotionTitle=Feeling symptoms?
2024-08-26 09:25:53,071 [ERROR] [MORE PROMOS] Error More Promotions
Traceback (most recent call last):
  File "D:\MS-Rewards-Farmer\src\morePromotions.py", line 39, in completeMorePromotions
    self.activities.openMorePromotionsActivity(
  File "D:\MS-Rewards-Farmer\src\activities.py", line 29, in openMorePromotionsActivity
    self.browser.utils.switchToNewTab(timeToWait=5)
  File "D:\MS-Rewards-Farmer\src\utils.py", line 216, in switchToNewTab
    self.webdriver.switch_to.window(window_name=self.webdriver.window_handles[1])
                                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^
IndexError: list index out of range
2024-08-26 09:25:55,126 [DEBUG] promotionTitle=Plan a quick getaway
2024-08-26 09:26:09,647 [DEBUG] promotionTitle=Plan a quick getaway
2024-08-26 09:26:10,119 [ERROR] [MORE PROMOS] Error More Promotions
Traceback (most recent call last):
  File "D:\MS-Rewards-Farmer\src\morePromotions.py", line 39, in completeMorePromotions
    self.activities.openMorePromotionsActivity(
  File "D:\MS-Rewards-Farmer\src\activities.py", line 29, in openMorePromotionsActivity
    self.browser.utils.switchToNewTab(timeToWait=5)
  File "D:\MS-Rewards-Farmer\src\utils.py", line 216, in switchToNewTab
    self.webdriver.switch_to.window(window_name=self.webdriver.window_handles[1])
                                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^
IndexError: list index out of range
2024-08-26 09:26:13,397 [DEBUG] promotionTitle=Stay in touch on hot topics!
2024-08-26 09:26:13,473 [ERROR] [MORE PROMOS] Error More Promotions
Traceback (most recent call last):
  File "D:\MS-Rewards-Farmer\src\morePromotions.py", line 39, in completeMorePromotions
    self.activities.openMorePromotionsActivity(
  File "D:\MS-Rewards-Farmer\src\activities.py", line 29, in openMorePromotionsActivity
    self.browser.utils.switchToNewTab(timeToWait=5)
  File "D:\MS-Rewards-Farmer\src\utils.py", line 216, in switchToNewTab
    self.webdriver.switch_to.window(window_name=self.webdriver.window_handles[1])
                                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^
IndexError: list index out of range
2024-08-26 09:26:15,025 [DEBUG] promotionTitle=Discover open job roles
2024-08-26 09:26:15,062 [ERROR] [MORE PROMOS] Error More Promotions
Traceback (most recent call last):
  File "D:\MS-Rewards-Farmer\src\morePromotions.py", line 39, in completeMorePromotions
    self.activities.openMorePromotionsActivity(
  File "D:\MS-Rewards-Farmer\src\activities.py", line 29, in openMorePromotionsActivity
    self.browser.utils.switchToNewTab(timeToWait=5)
  File "D:\MS-Rewards-Farmer\src\utils.py", line 216, in switchToNewTab
    self.webdriver.switch_to.window(window_name=self.webdriver.window_handles[1])
                                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^
IndexError: list index out of range
2024-08-26 09:26:17,570 [DEBUG] promotionTitle=Earn More Rewards points on mobile
2024-08-26 09:26:17,570 [DEBUG] Already done, continuing
2024-08-26 09:26:17,570 [DEBUG] promotionTitle=Time for some new shoes?
2024-08-26 09:26:17,570 [DEBUG] Already done, continuing
2024-08-26 09:26:17,570 [DEBUG] promotionTitle=Do you know the answer?
2024-08-26 09:26:17,571 [DEBUG] Already done, continuing
2024-08-26 09:26:21,072 [DEBUG] in __exit__ exc_type=<class 'AssertionError'> exc_value= traceback=<traceback object at 0x000001EDF5487380>
2024-08-26 09:26:22,765 [ERROR]
Traceback (most recent call last):
  File "D:\MS-Rewards-Farmer\main.py", line 40, in main
    earned_points = executeBot(currentAccount, args)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\MS-Rewards-Farmer\main.py", line 245, in executeBot
    MorePromotions(desktopBrowser).completeMorePromotions()
  File "D:\MS-Rewards-Farmer\src\morePromotions.py", line 123, in completeMorePromotions
    Utils.sendNotification("Incomplete promotions(s)", incompletePromotions)
  File "D:\MS-Rewards-Farmer\src\utils.py", line 69, in sendNotification
    assert apprise.notify(title=str(title), body=str(body))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError
2024-08-26 09:26:22,782 [ERROR]
Traceback (most recent call last):
  File "D:\MS-Rewards-Farmer\main.py", line 40, in main
    earned_points = executeBot(currentAccount, args)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\MS-Rewards-Farmer\main.py", line 245, in executeBot
    MorePromotions(desktopBrowser).completeMorePromotions()
  File "D:\MS-Rewards-Farmer\src\morePromotions.py", line 123, in completeMorePromotions
    Utils.sendNotification("Incomplete promotions(s)", incompletePromotions)
  File "D:\MS-Rewards-Farmer\src\utils.py", line 69, in sendNotification
    assert apprise.notify(title=str(title), body=str(body))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "D:\MS-Rewards-Farmer\main.py", line 356, in <module>
    main()
  File "D:\MS-Rewards-Farmer\main.py", line 43, in main
    Utils.sendNotification(
  File "D:\MS-Rewards-Farmer\src\utils.py", line 69, in sendNotification
    assert apprise.notify(title=str(title), body=str(body))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError
Traceback (most recent call last):
  File "D:\MS-Rewards-Farmer\main.py", line 40, in main
    earned_points = executeBot(currentAccount, args)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\MS-Rewards-Farmer\main.py", line 245, in executeBot
    MorePromotions(desktopBrowser).completeMorePromotions()
  File "D:\MS-Rewards-Farmer\src\morePromotions.py", line 123, in completeMorePromotions
    Utils.sendNotification("Incomplete promotions(s)", incompletePromotions)
  File "D:\MS-Rewards-Farmer\src\utils.py", line 69, in sendNotification
    assert apprise.notify(title=str(title), body=str(body))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "D:\MS-Rewards-Farmer\main.py", line 356, in <module>
    main()
  File "D:\MS-Rewards-Farmer\main.py", line 43, in main
    Utils.sendNotification(
  File "D:\MS-Rewards-Farmer\src\utils.py", line 69, in sendNotification
    assert apprise.notify(title=str(title), body=str(body))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "D:\MS-Rewards-Farmer\main.py", line 359, in <module>
    Utils.sendNotification(
  File "D:\MS-Rewards-Farmer\src\utils.py", line 69, in sendNotification
    assert apprise.notify(title=str(title), body=str(body))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError`

--- a/main.py
+++ b/main.py
@@ -26,20 +26,20 @@ def main():
     # Load previous day's points data
     previous_points_data = load_previous_points_data()
 
+    foundError = False
+
     for currentAccount in CONFIG.accounts:
         try:
             earned_points = executeBot(currentAccount)
         except Exception as e1:
             logging.error("", exc_info=True)
+            foundError = True
             if CONFIG.get("apprise.notify.uncaught-exception"):
                 APPRISE.notify(
                     f"{type(e1).__name__}: {e1}",
                     f"⚠️ Error executing {currentAccount.email}, please check the log",
                 )
-            if len(CONFIG.accounts) > 1:
-                continue
-            else:
-                exit(1)
+            continue
 
         previous_points = previous_points_data.get(currentAccount.email, 0)
 
@@ -59,6 +59,9 @@ def main():
     # Save the current day's points data for the next day in the "logs" folder
     save_previous_points_data(previous_points_data)
     logging.info("[POINTS] Data saved for the next day.")
+
+    if foundError:
+        exit(1)
 
 
 def log_daily_points_to_csv(earned_points, points_difference):

--- a/main.py
+++ b/main.py
@@ -4,7 +4,6 @@ import logging
 import logging.config
 import logging.handlers as handlers
 import sys
-import traceback
 from datetime import datetime
 from enum import Enum, auto
 
@@ -18,7 +17,7 @@ from src import (
 from src.activities import Activities
 from src.browser import RemainingSearches
 from src.loggingColoredFormatter import ColoredFormatter
-from src.utils import CONFIG, sendNotification, getProjectRoot, formatNumber
+from src.utils import CONFIG, APPRISE, getProjectRoot, formatNumber
 
 
 def main():
@@ -32,12 +31,16 @@ def main():
             earned_points = executeBot(currentAccount)
         except Exception as e1:
             logging.error("", exc_info=True)
-            sendNotification(
-                f"⚠️ Error executing {currentAccount.email}, please check the log",
-                traceback.format_exc(),
-                e1,
-            )
-            continue
+            if CONFIG.get("apprise.notify.uncaught-exception"):
+                APPRISE.notify(
+                    f"{type(e1).__name__}: {e1}",
+                    f"⚠️ Error executing {currentAccount.email}, please check the log",
+                )
+            if len(CONFIG.accounts) > 1:
+                continue
+            else:
+                exit(1)
+
         previous_points = previous_points_data.get(currentAccount.email, 0)
 
         # Calculate the difference in points from the prior day
@@ -200,8 +203,7 @@ def executeBot(currentAccount):
                 f" ({goalTitle})"
             )
 
-        sendNotification(
-            "Daily Points Update",
+        APPRISE.notify(
             "\n".join(
                 [
                     f"👤 Account: {currentAccount.email}",
@@ -210,12 +212,13 @@ def executeBot(currentAccount):
                     goalStatus,
                 ]
             ),
+            "Daily Points Update",
         )
     elif appriseSummary == AppriseSummary.ON_ERROR:
         if remainingSearches.getTotal() > 0:
-            sendNotification(
-                "Error: remaining searches",
+            APPRISE.notify(
                 f"account email: {currentAccount.email}, {remainingSearches}",
+                "Error: remaining searches",
             )
     elif appriseSummary == AppriseSummary.NEVER:
         pass
@@ -259,7 +262,9 @@ if __name__ == "__main__":
         main()
     except Exception as e:
         logging.exception("")
-        sendNotification(
-            "⚠️ Error occurred, please check the log", traceback.format_exc(), e
-        )
+        if CONFIG.get("apprise.notify.uncaught-exception"):
+            APPRISE.notify(
+                f"{type(e).__name__}: {e}",
+                "⚠️ Error occurred, please check the log",
+            )
         exit(1)

--- a/src/activities.py
+++ b/src/activities.py
@@ -9,7 +9,7 @@ from selenium.webdriver.remote.webelement import WebElement
 
 from src.browser import Browser
 from src.constants import REWARDS_URL
-from src.utils import CONFIG, sendNotification, getAnswerCode, cooldown
+from src.utils import CONFIG, APPRISE, getAnswerCode, cooldown
 
 
 class Activities:
@@ -215,9 +215,9 @@ class Activities:
                     incompleteActivities.append(activityTitle)
             if incompleteActivities:
                 logging.info(f"incompleteActivities: {incompleteActivities}")
-                sendNotification(
-                    f"We found some incomplete activities for {self.browser.email}",
+                APPRISE.notify(
                     '"' + '", "'.join(incompleteActivities) + '"\n' + REWARDS_URL,
+                    f"We found some incomplete activities for {self.browser.email}",
                 )
 
 

--- a/src/login.py
+++ b/src/login.py
@@ -1,7 +1,5 @@
-import argparse
 import contextlib
 import logging
-from argparse import Namespace
 
 from pyotp import TOTP
 from selenium.common import TimeoutException
@@ -13,7 +11,7 @@ from selenium.webdriver.common.by import By
 from undetected_chromedriver import Chrome
 
 from src.browser import Browser
-from src.utils import sendNotification, CONFIG
+from src.utils import CONFIG, APPRISE
 
 
 class Login:
@@ -101,8 +99,10 @@ class Login:
                 codeField.text,
             )
             if CONFIG.get("apprise.notify.login-code"):
-                sendNotification(
-                    f"Confirm your login on your phone", f"Code: {codeField.text} (expires in 1 minute)")
+                APPRISE.notify(
+                    f"Code: {codeField.text} (expires in 1 minute)",
+                    f"Confirm your login on your phone",
+                )
             self.utils.waitUntilVisible(By.NAME, "kmsiForm", 60)
             logging.info("[LOGIN] Successfully verified!")
         else:
@@ -143,8 +143,10 @@ class Login:
                     codeField.text,
                 )
                 if CONFIG.get("apprise.notify.login-code"):
-                    sendNotification(
-                        f"Confirm your login on your phone", f"Code: {codeField.text} (expires in 1 minute)")
+                    APPRISE.notify(
+                        f"Code: {codeField.text} (expires in 1 minute)",
+                        f"Confirm your login on your phone",
+                    )
                 self.utils.waitUntilVisible(By.NAME, "kmsiForm", 60)
                 logging.info("[LOGIN] Successfully verified!")
 

--- a/src/searches.py
+++ b/src/searches.py
@@ -2,8 +2,7 @@ import dbm.dumb
 import logging
 import shelve
 from enum import Enum, auto
-from itertools import cycle
-from random import random, randint, shuffle
+from random import random, randint
 from time import sleep
 from typing import Final
 

--- a/src/utils.py
+++ b/src/utils.py
@@ -560,24 +560,26 @@ def loadConfig(configFilename="config.yaml") -> Config:
         createEmptyConfig(configFile, args_config)
 
     config = DEFAULT_CONFIG | Config.fromYaml(configFile) | args_config
-    config = setupAccounts(config)
+
+    if config.rtfr:
+        print("Please read the README.md file before using this script. Exiting.")
+        exit(0)
 
     return config
 
 
-def sendNotification(title: str, body: str, e: Exception = None) -> None:
-    if not CONFIG.apprise.enabled or (
-        e and not CONFIG.get("apprise.notify.uncaught-exception")
-    ):
-        return
+def initApprise() -> Apprise:
     apprise = Apprise()
-    urls: list[str] = CONFIG.apprise.urls
+
+    urls = []
+    if CONFIG.apprise.enabled:
+        urls: list[str] = CONFIG.apprise.urls
+
     if not urls:
         logging.debug("No urls found, not sending notification")
-        return
-    for url in urls:
-        apprise.add(url)
-    assert apprise.notify(title=str(title), body=str(body))
+
+    apprise.add(urls)
+    return apprise
 
 
 def getAnswerCode(key: str, string: str) -> str:
@@ -629,8 +631,5 @@ def cooldown() -> None:
     logging.info(f"[COOLDOWN] Waiting for {cooldownTime} seconds")
     time.sleep(cooldownTime)
 
-
 CONFIG = loadConfig()
-if CONFIG.rtfr:
-    print("Please read the README.md file before using this script. Exiting.")
-    exit(0)
+APPRISE = initApprise()

--- a/src/utils.py
+++ b/src/utils.py
@@ -574,9 +574,8 @@ def initApprise() -> Apprise:
     urls = []
     if CONFIG.apprise.enabled:
         urls: list[str] = CONFIG.apprise.urls
-
-    if not urls:
-        logging.debug("No urls found, not sending notification")
+        if not urls:
+            logging.info("No apprise urls found, not sending notification")
 
     apprise.add(urls)
     return apprise

--- a/test/test_main.py
+++ b/test/test_main.py
@@ -2,7 +2,6 @@ import unittest
 from unittest.mock import patch, MagicMock
 
 import main
-from src import utils
 from src.utils import Config, CONFIG
 
 

--- a/test/test_utils.py
+++ b/test/test_utils.py
@@ -1,9 +1,9 @@
 from unittest import TestCase
 
-from src.utils import CONFIG, sendNotification
+from src.utils import CONFIG, APPRISE
 
 
 class TestUtils(TestCase):
     def test_send_notification(self):
         CONFIG.apprise.enabled = True
-        sendNotification("title", "body")
+        APPRISE.notify("body", "title")

